### ISSUE TITLE
egapx:rnaseq_long_plane:minimap2:minimap2_wnode is running very long time and don't complete

### ISSUE THREAD

**Issue by sidevshiy** [ncbi/egapx#126]
Hello EGAPx team,

Thank you for providing such a useful annotation pipeline. I’m running the 0.4.0-alpha release on a wheat (hexaploid) genome with custom Illumina and PacBio reads and have run into two issues:

 1. rnaseq_long_plane:minimap2_wnode jobs stall after two successes
The module launches 155 jobs, but only the first two finish; the remaining 153 stay in RUNNING indefinitely and are never marked complete. A typical log fragment looks like this

03005/001/0582/RE 659E0BBD84A88341 1199/1151 2025-06-12T08:25:19.006640 warlock34       UNK_CLIENT      UNK_SESSION              minimap2_wnode request-stop  200 0.000062942 0 0
  03005/001/0583/RB 659E0BBD84A88341 1200/1152 2025-06-12T08:25:22.007065 warlock34       UNK_CLIENT      UNK_SESSION              minimap2_wnode request-start internal_thread=monitor&cpu_count=7&normalized_load=59&total_ram=810803957760&effective_ram=808656474112&pct_ram_used=0&mem_total=453656576&mem_peak=453656576&mem_self=453656576&mem_available=808202817536&ncbi_app_version=0.0.27834&ncbi_app_sc_version=29&ncbi_app_vcs_revision=698072&ncbi_app_revision=698072
  03005/001/0583/RE 659E0BBD84A88341 1201/1153 2025-06-12T08:25:22.007111 warlock34       UNK_CLIENT      UNK_SESSION              minimap2_wnode request-stop  200 0.000063180 0 0
  03005/001/0584/RB 659E0BBD84A88341 1202/1154 2025-06-12T08:25:25.007530 warlock34       UNK_CLIENT      UNK_SESSION              minimap2_wnode request-start internal_thread=monitor&cpu_count=7&normalized_load=59&total_ram=810803957760&effective_ram=808656474112&pct_ram_used=0&mem_total=453656576&mem_peak=453656576&mem_self=453656576&mem_available=808202817536&ncbi_app_version=0.0.27834&ncbi_app_sc_version=29&ncbi_app_vcs_revision=698072&ncbi_app_revision=698072
  03005/001/0584/RE 659E0BBD84A88341 1203/1155 2025-06-12T08:25:25.007574 warlock34       UNK_CLIENT      UNK_SESSION              minimap2_wnode request-stop  200 0.000061988 0 0
  03005/001/0585/RB 659E0BBD84A88341 1204/1156 2025-06-12T08:25:28.008038 warlock34       UNK_CLIENT      UNK_SESSION              minimap2_wnode request-start internal_thread=monitor&cpu_count=7&normalized_load=59&total_ram=810803957760&effective_ram=808656474112&pct_ram_used=0&mem_total=453656576&mem_peak=453656576&mem_self=453656576&mem_available=808202817536&ncbi_app_version=0.0.27834&ncbi_app_sc_version=29&ncbi_app_vcs_revision=698072&ncbi_app_revision=698072
  03005/001/0585/RE 659E0BBD84A88341 1205/1157 2025-06-12T08:25:28.008083 warlock34       UNK_CLIENT      UNK_SESSION              minimap2_wnode request-stop  200 0.000062942 0 0
  03005/001/0586/RB 659E0BBD84A88341 1206/1158 2025-06-12T08:25:31.008487 warlock34       UNK_CLIENT      UNK_SESSION              minimap2_wnode request-start internal_thread=monitor&cpu_count=7&normalized_load=60&total_ram=810803957760&effective_ram=808656474112&pct_ram_used=0&mem_total=453656576&mem_peak=453656576&mem_self=453656576&mem_available=808202817536&ncbi_app_version=0.0.27834&ncbi_app_sc_version=29&ncbi_app_vcs_revision=698072&ncbi_app_revision=698072
  03005/001/0586/RE 659E0BBD84A88341 1207/1159 2025-06-12T08:25:31.008532 warlock34       UNK_CLIENT      UNK_SESSION              minimap2_wnode request-stop  200 0.000064134 0 0
  03005/001/0587/RB 659E0BBD84A88341 1208/1160 2025-06-12T08:25:34.008972 warlock34       UNK_CLIENT      UNK_SESSION              minimap2_wnode request-start internal_thread=monitor&cpu_count=7&normalized_load=60&total_ram=810803957760&effective_ram=808656474112&pct_ram_used=0&mem_total=453656576&mem_peak=453656576&mem_self=453656576&mem_available=808202817536&ncbi_app_version=0.0.27834&ncbi_app_sc_version=29&ncbi_app_vcs_revision=698072&ncbi_app_revision=698072
  03005/001/0587/RE 659E0BBD84A88341 1209/1161 2025-06-12T08:25:34.009014 warlock34       UNK_CLIENT      UNK_SESSION              minimap2_wnode request-stop  200 0.000060796 0 0

Is this behaviour expected (maybe I just need to wait more), or is there a configuration problem on my side?
If not expected, could you advise on the correct resource settings or
debug steps?

During the STAR genome-index step I get: please specify --limitGenomeGenerateRAM
I added --limitGenomeGenerateRAM through config file, but the error persists. I  noticed two STAR calls:

script:
        out_dir = genome_file.toString().replaceFirst(/\.(fa(sta)?|fna)$/, ".index")
    """
    mkdir -p $out_dir
    echo "in $genome_file, out $out_dir"
    # Check that --genomeSAindexNbases is not set
    if [[ ! "$parameters" =~ --genomeSAindexNbases ]]; then
        # Formula from https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf, section 2.2.5
        sa_index_bases=`python3 -c "import os;import math;print(math.floor(min(14, math.log2(os.path.getsize('$genome_file'))/2 - 1)))"`
        effective_parameters="$parameters --genomeSAindexNbases \$sa_index_bases"
    else
        effective_parameters="$parameters"
    fi
    echo "ls"
    ls

    echo "STAR index effective parameters: \$effective_parameters"
    STAR  \$effective_parameters --runMode genomeGenerate --genomeDir $out_dir --genomeFastaFiles $genome_file
    chmod a+rx $out_dir
    STAR  $parameters --runMode genomeGenerate --genomeDir $out_dir --genomeFastaFiles $genome_file
    """

I suspect the second call re-runs without the adjusted parameters and
triggers the same memory check. Is this duplication intentional? If
not, which block should be removed or guarded by an if?

**Comment by sidevshiy** [comment]
Noticed that in .command.log I have the following error.

 Info: SeqId chunk file asncache/seq_id_chunk does not exist.  Creating.
#canonical-id
Chunk file asncache/chunk.00001 does not exist.  Creating.
lcl|Chr1A_RagTag
lcl|Chr1B_RagTag
lcl|Chr1D_RagTag
lcl|Chr2A_RagTag
lcl|Chr2B_RagTag
lcl|Chr2D_RagTag
lcl|Chr3A_RagTag
lcl|Chr3B_RagTag
lcl|Chr3D_RagTag
lcl|Chr4A_RagTag
lcl|Chr4B_RagTag
lcl|Chr4D_RagTag
lcl|Chr5A_RagTag
lcl|Chr5B_RagTag
lcl|Chr5D_RagTag
lcl|Chr6A_RagTag
lcl|Chr6B_RagTag
lcl|Chr6D_RagTag
lcl|Chr7A_RagTag
lcl|Chr7B_RagTag
lcl|Chr7D_RagTag
lcl|Chr0_RagTag
done, dumped 22 items
     Stack trace:
      /img/gp/arch/x86_64/lib/libxncbi.so /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/src/corelib/ncbidiag.cpp:5770 ncbi::SDiagMessage::x_OldWrite(std::ostream&, int) const offset=0x0 addr=0x7fe7db997a7c
       /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/src/corelib/ncbidiag.cpp:6485 operator<< offset=0x0 addr=(nil)
      /img/gp/arch/x86_64/lib/libxncbi.so /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/include/corelib/ncbidiag.hpp:1763 ncbi::CStreamDiagHandler::Post(ncbi::SDiagMessage const&) offset=0x0 addr=0x7fe7db998be2
      /img/gp/arch/x86_64/lib/libxncbi.so /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/src/corelib/ncbidiag.cpp:4178 ncbi::CDiagBuffer::DiagHandler(ncbi::SDiagMessage&) offset=0x0 addr=0x7fe7db98b38a
      /img/gp/arch/x86_64/lib/libxncbi.so /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/src/corelib/ncbidiag.cpp:4412 ncbi::CDiagBuffer::Flush() offset=0x0 addr=0x7fe7db98218f
       /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/include/corelib/ncbidiag.inl:497 Detach offset=0x0 addr=(nil)
       /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/include/corelib/ncbidiag.inl:411 EndMess offset=0x0 addr=(nil)
      /img/gp/arch/x86_64/bin/prime_cache /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/include/corelib/ncbidiag.inl:505 ncbi::Endm(ncbi::CNcbiDiag const&) [clone .isra.0] offset=0x0 addr=0x413305
       /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/src/app/asn_cache/prime_cache.cpp:667 operator<< offset=0x0 addr=(nil)
      /img/gp/arch/x86_64/bin/prime_cache /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/include/corelib/ncbidiag.hpp:1017 CPrimeCacheApplication::x_Process_Fasta(std::istream&, std::ostream&) offset=0x0 addr=0x41985a
      /img/gp/arch/x86_64/bin/prime_cache /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/src/app/asn_cache/prime_cache.cpp:1345 CPrimeCacheApplication::Run() offset=0x0 addr=0x41f8ad
      /img/gp/arch/x86_64/lib/libxncbi.so /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/src/corelib/ncbiapp.cpp:737 ncbi::CNcbiApplicationAPI::x_TryMain(ncbi::EAppDiagStream, char const*, int*, bool*) offset=0x0 addr=0x7fe7db947852
      /img/gp/arch/x86_64/lib/libxncbi.so /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/src/corelib/ncbiapp.cpp:1083 ncbi::CNcbiApplicationAPI::AppMain(int, char const* const*, char const* const*, ncbi::EAppDiagStream, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) offset=0x0 addr=0x7fe7db949f27
      /img/gp/arch/x86_64/bin/prime_cache /export/home/gpipe/TeamCity/Agent2/work/6d4fb99b9e4908f7/ncbi_cxx/src/app/asn_cache/prime_cache.cpp:1411 main offset=0x0 addr=0x412ecb
      /usr/lib/x86_64-linux-gnu/libc.so.6 :0  offset=0x0 addr=0x7fe7d9a181c9
      /usr/lib/x86_64-linux-gnu/libc.so.6 :0  offset=0x0 addr=0x7fe7d9a1828a
      /img/gp/arch/x86_64/bin/prime_cache :0  offset=0x0 addr=0x41307d
       :0  offset=0x0 addr=0xffffffffffffffff

03770/000/0000/PB 659F0EBA84B410B1 0001/0001 2025-06-12T21:05:15.579213 warlock35       UNK_CLIENT      UNK_SESSION              minimap2_wnode Info: CORELIB(106.11) "ncbiapp.cpp", line 1271: ncbi::CNcbiApplicationAPI::LoadConfig() --- Registry file of application "minimap2_wnode" is not found
03770/000/0000/PB 659F0EBA84B410B1 0002/0002 2025-06-12T21:05:15.582033 warlock35       UNK_CLIENT      UNK_SESSION              minimap2_wnode start         /img/gp/bin/minimap2_wnode -minimap2-executable /img/gp/bin/minimap2 -filter-executable /img/gp/bin/exon_selector -start-job-id 9 -input-jobs job.10 -nogenbank -lds2 GF361_iso_seq_2.LDS -asn-cache asncache -gc GF361-gencoll.asn -work-area tmp -O interim -max-intron 300000 -est_colocation_weight 0 -min-entropy 0.65 -workers 1 -generate-stats -output_one_orientation_only -trust_input_collation -with-best-placement -minimap2-params '-x splice --eqx --MD --cs -Y --sam-hit-only -M 0.9 -p 0.1 -N 25'
03770/000/0000/PB 659F0EBA84B410B1 0003/0003 2025-06-12T21:05:15.582498 warlock35       UNK_CLIENT      UNK_SESSION              minimap2_wnode extra         ncbi_app_username=liakh&ncbi_app_path=/img/gp/bin/minimap2_wnode&ncbi_app_build_date=Jun++1+2025+06:27:04&ncbi_app_tc_project=Software+Compilation+%26+Artifact+Generation&ncbi_app_tc_conf=Release&ncbi_app_tc_build=27834&ncbi_app_build_id=55421117&ncbi_app_built_as=minimap2_wnode&ncbi_app_version=0.0.27834&ncbi_app_sc_version=29&ncbi_app_vcs_revision=698072&ncbi_app_revision=698072
03770/000/0000/P  659F0EBA84B410B1 0004/0004 2025-06-12T21:05:15.594675 warlock35       UNK_CLIENT      UNK_SESSION              minimap2_wnode Note[I]: DATA_LOADERS_UTIL "data_loaders_util.cpp", line 400: ncbi::objects::CDataLoadersUtil::x_SetupLDS2DataLoader() --- added loader: LDS2: /fastscratch/liakh/nextflow/egapx/GF361/nf_work/a6/e348073215fcf07818154356bc47b9/GF361_iso_seq_2.LDS (10)
03770/000/0000/P  659F0EBA84B410B1 0005/0005 2025-06-12T21:05:15.594763 warlock35       UNK_CLIENT      UNK_SESSION              minimap2_wnode Note[I]: DATA_LOADERS_UTIL "data_loaders_util.cpp", line 438: ncbi::objects::CDataLoadersUtil::x_SetupASNCacheDataLoader() --- added loader: ASNCache: asncache (11)

**Comment by sidevshiy** [comment]
My fault. I provided hifi reads of isoseq instead of flnc reads. That is why it was longer. I will close the issue
